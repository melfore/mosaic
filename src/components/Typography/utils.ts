type VariantDefaultsType = {
  [key: string]: {
    bottomSpacing: boolean;
    truncated: boolean;
  };
};

const VARIANT_DEFAULTS: VariantDefaultsType = {
  body1: {
    bottomSpacing: false,
    truncated: false,
  },
  caption: {
    bottomSpacing: false,
    truncated: false,
  },
  h5: {
    bottomSpacing: true,
    truncated: true,
  },
  h6: {
    bottomSpacing: true,
    truncated: true,
  },
  overline: {
    bottomSpacing: false,
    truncated: false,
  },
  subtitle1: {
    bottomSpacing: false,
    truncated: false,
  },
};

export const getBottomSpacing = (variant: string, bottomSpacing?: boolean): boolean =>
  bottomSpacing === undefined ? VARIANT_DEFAULTS[variant].bottomSpacing : bottomSpacing;

export const getTruncate = (variant: string, truncated?: boolean): boolean =>
  truncated === undefined ? VARIANT_DEFAULTS[variant].truncated : truncated;

export const VARIANT_COMPONENT_MAP = {
  body1: "p",
  caption: "span",
  h5: "h1",
  h6: "h2",
  overline: "h4",
  subtitle1: "h3",
};
