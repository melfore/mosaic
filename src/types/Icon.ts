import { LoadableType } from "./Base";

export enum Icons {
  account = "account",
  add = "add",
  apps = "apps",
  block = "block",
  brightness = "brightness",
  business = "business",
  chart_timeline = "chart_timeline",
  check = "check",
  checkbox = "checkbox",
  checkbox_empty = "checkbox_empty",
  clock = "clock",
  close = "close",
  delete = "delete",
  down = "down",
  edit = "edit",
  error = "error",
  filter = "filter",
  first = "first",
  home = "home",
  last = "last",
  left = "left",
  list = "list",
  list_check = "list_check",
  mail = "mail",
  mail_open = "mail_open",
  menu = "menu",
  notifications = "notifications",
  open_new = "open_new",
  pause = "pause",
  play = "play",
  print = "print",
  refresh = "refresh",
  right = "right",
  save = "save",
  search = "search",
  send = "send",
  settings = "settings",
  up = "up",
  up_arrow = "up_arrow",
  warn = "warn",
}

export enum IconSize {
  default = "default",
  large = "large",
  small = "small",
}

interface IconForwardedType {
  className?: string;
  color?: string;
  ref?: any;
}

export interface IconType extends LoadableType {
  forwarded?: IconForwardedType;
  name: Icons;
  size?: IconSize;
}
