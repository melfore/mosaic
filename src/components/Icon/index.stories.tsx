import React from 'react';
import {  } from '@storybook/addon-actions';
import { select } from '@storybook/addon-knobs';
import { Icons, IconSize } from '../../types/Icon';
import { DOCS_PAGE_STRUCTURE, StoriesWrapper } from '../../utils/storybook';
import Icon from '.';

export default {
  title: 'Icon',
  component: Icon,
  parameters: {
    ...DOCS_PAGE_STRUCTURE,
  },
}

export const Canvas = () => (
  <Icon
    dataCy="icon"
    name={select('icon', Icons, Icons.add)}
    size={select('size', IconSize, IconSize.default)}
  />
);

export const Size = () => (
  <StoriesWrapper>
    <Icon
      dataCy="icon-send"
      name={Icons.send}
      size={IconSize.small}
    />
    <Icon
      dataCy="icon-send"
      name={Icons.send}
    />
    <Icon
      dataCy="icon-send"
      name={Icons.send}
      size={IconSize.large}
    />
  </StoriesWrapper>
);

const allIcons = Object
  .keys(Icons)
  .map((icon: string) => (
    <div className="icon-wrapper">
      <Icon
        dataCy={`icon-${icon}`}
        name={Icons[icon]}
      />
      <span>
        {icon}
      </span>
    </div>
  ));

export const AllIcons = () => (
  <StoriesWrapper>
    {allIcons}
  </StoriesWrapper>
);
