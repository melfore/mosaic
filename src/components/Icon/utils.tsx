import React from "react";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUISendIcon from "@material-ui/icons/Send";
import { Icons } from "../../types/Icon";

export const iconsCatalog = {
  [Icons.add]: <MUIAddIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.playlistAddCheck]: <MUIPlaylistAddCheckIcon />,
  [Icons.send]: <MUISendIcon />,
};
