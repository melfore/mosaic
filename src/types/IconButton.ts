import { MouseEvent } from "react";
import { Icons, IconSize } from "./Icon";
import { BaseType } from "./Base";

export interface IconButtonType extends BaseType {
  icon: Icons;
  onClick: (event: MouseEvent) => void;
  disabled?: boolean;
  size?: IconSize;
}
