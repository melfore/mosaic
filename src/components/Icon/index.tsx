import React, { FC } from "react";
import { IconType, IconSize } from "../../types/Icon";
import { iconsCatalog } from "./utils";

const Icon: FC<IconType> = ({ dataCy, forwarded = {}, name, size = IconSize.default }) => {
  const icon = iconsCatalog[name];
  return React.cloneElement(icon, { ...forwarded, "data-cy": dataCy, fontSize: size });
};

export default Icon;
