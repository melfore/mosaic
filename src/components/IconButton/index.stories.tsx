import React from "react";
import { action } from "@storybook/addon-actions";
import { select, text } from "@storybook/addon-knobs";
import { Icons, IconSize } from "../../types/Icon";
import { getDocsPageStructure, StoriesWrapper } from "../../utils/stories";
import IconButton from ".";

export default {
  title: "IconButton",
  component: IconButton,
  parameters: {
    ...getDocsPageStructure("IconButton", false),
  },
};

export const Canvas = () => (
  <IconButton
    dataCy={text("data-cy", "icon-button-identifier")}
    icon={select("icon", Icons, Icons.add)}
    onClick={action("Click on IconButton")}
    size={select("size", IconSize, IconSize.default)}
  />
);

export const Size = () => (
  <StoriesWrapper>
    <IconButton icon={Icons.add} size={IconSize.small} onClick={() => {}} />
    <IconButton icon={Icons.add} onClick={() => {}} />
    <IconButton icon={Icons.add} size={IconSize.large} onClick={() => {}} />
  </StoriesWrapper>
);

export const Disabled = () => (
  <StoriesWrapper>
    <IconButton icon={Icons.add} onClick={() => {}} />
    <IconButton icon={Icons.add} onClick={() => {}} disabled />
  </StoriesWrapper>
);
