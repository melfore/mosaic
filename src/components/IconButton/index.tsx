import React, { FC } from "react";
import MUIIconButton from "@material-ui/core/IconButton";
import Icon from "../Icon";
import { IconButtonType } from "../../types/IconButton";
import { IconSize } from "../../types/Icon";
import { Color } from "../../types/Base";

/**
 * IconButton component made on top of `@material-ui/core/IconButton`.
 *
 * Add required documentation, use HTML tags if needed.
 */
const IconButton: FC<IconButtonType> = ({
  color = Color.primary,
  dataCy = "icon-button",
  icon,
  onClick,
  disabled = false,
  size = IconSize.default,
}) => {
  const _icon = <Icon dataCy={`${dataCy}`} name={icon} size={size} />;

  return (
    <MUIIconButton color={color} onClick={onClick} disabled={disabled}>
      {_icon}
    </MUIIconButton>
  );
};

export default IconButton;
