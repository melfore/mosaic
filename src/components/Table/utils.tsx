import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionScope, TableActionType, TableColumnType } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: TableActionType): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    scope = TableActionScope.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    isFreeAction: scope === TableActionScope.default,
    onClick: callback,
  };
};

export const actionComponentAdapter = (props: any) => {
  const { data, action } = props;
  const { disabled, hidden, icon, isFreeAction, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {isFreeAction && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
          />
        </Fragment>
      )}
      {!isFreeAction && (
        <Fragment>
          <IconButton
            disabled={disabled}
            icon={iconName}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: TableColumnType): MTColumnType<object> => {
  const { label, path, render } = column;
  return {
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
