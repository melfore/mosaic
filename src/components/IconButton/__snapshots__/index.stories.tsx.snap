// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs/IconButton Badge smoke-test 1`] = `
<span data-cy="icon-button-adornment">
  <span data-cy="icon-button-adornment-badge"
        class="MuiBadge-root css-mui-classname"
  >
    <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
            tabindex="0"
            type="button"
            data-cy="icon-button"
    >
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="AddIcon"
           data-cy="icon-button-icon"
      >
        <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z">
        </path>
      </svg>
    </button>
    <span class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightCircular MuiBadge-overlapCircular MuiBadge-colorSecondary css-mui-classname">
      8
    </span>
  </span>
</span>
`;

exports[`Inputs/IconButton CustomIcon smoke-test 1`] = `
<button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
        tabindex="0"
        type="button"
        data-cy="icon-button"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="StyleIcon"
       data-cy="icon-button-icon"
       style="height: 24px; width: 24px;"
  >
    <path d="m2.53 19.65 1.34.56v-9.03l-2.43 5.86c-.41 1.02.08 2.19 1.09 2.61m19.5-3.7L17.07 3.98c-.31-.75-1.04-1.21-1.81-1.23-.26 0-.53.04-.79.15L7.1 5.95c-.75.31-1.21 1.03-1.23 1.8-.01.27.04.54.15.8l4.96 11.97c.31.76 1.05 1.22 1.83 1.23.26 0 .52-.05.77-.15l7.36-3.05c1.02-.42 1.51-1.59 1.09-2.6M7.88 8.75c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1m-2 11c0 1.1.9 2 2 2h1.45l-3.45-8.34z">
    </path>
  </svg>
</button>
`;

exports[`Inputs/IconButton Disabled smoke-test 1`] = `
<button class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
        tabindex="-1"
        type="button"
        disabled
        data-cy="icon-button"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="AddIcon"
       data-cy="icon-button-icon"
  >
    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z">
    </path>
  </svg>
</button>
`;

exports[`Inputs/IconButton Primary play-test 1`] = `
<button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
        tabindex="0"
        type="button"
        data-cy="icon-button"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="AddIcon"
       data-cy="icon-button-icon"
  >
    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z">
    </path>
  </svg>
</button>
`;

exports[`Inputs/IconButton Rotate smoke-test 1`] = `
<button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
        tabindex="0"
        type="button"
        data-cy="icon-button"
>
  <div class="MuiBox-root css-mui-classname">
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="AddIcon"
         data-cy="icon-button-icon"
    >
      <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z">
      </path>
    </svg>
  </div>
</button>
`;

exports[`Inputs/IconButton SizeLarge smoke-test 1`] = `
<button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
        tabindex="0"
        type="button"
        data-cy="icon-button"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-mui-classname"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="AddIcon"
       data-cy="icon-button-icon"
  >
    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z">
    </path>
  </svg>
</button>
`;

exports[`Inputs/IconButton SizeSmall smoke-test 1`] = `
<button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
        tabindex="0"
        type="button"
        data-cy="icon-button"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-mui-classname"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="AddIcon"
       data-cy="icon-button-icon"
  >
    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z">
    </path>
  </svg>
</button>
`;

exports[`Inputs/IconButton Styled smoke-test 1`] = `
<button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
        tabindex="0"
        type="button"
        data-cy="icon-button"
        style="background-color: red; color: white;"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="AddIcon"
       data-cy="icon-button-icon"
  >
    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z">
    </path>
  </svg>
</button>
`;

exports[`Inputs/IconButton Tooltip smoke-test 1`] = `
<span data-cy="icon-button-adornment">
  <span aria-label="add"
        data-cy="icon-button-adornment-tooltip"
        class
  >
    <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
            tabindex="0"
            type="button"
            data-cy="icon-button"
    >
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="AddIcon"
           data-cy="icon-button-icon"
      >
        <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z">
        </path>
      </svg>
    </button>
  </span>
</span>
`;
